keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ExceptionHandler.kt

```diff
@@ -58,12 +77,121 @@ class ExceptionHandler {
     return ApiError(e)
   }
 
-  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+  private fun JsonMappingException.toDetails(): ParsingErrorDetails {
+    val (rootCause, problemPath) = findRootCause()
+
+    return ParsingErrorDetails(
+      error = when (this) {
+        // caters to missing properties at the root-level (e.g. serviceAccount)
+        is MissingKotlinParameterException -> ParsingError.MISSING_PROPERTY
+        else -> when (rootCause) {
+          null -> ParsingError.OTHER
+          is MissingKotlinParameterException -> ParsingError.MISSING_PROPERTY
+          is IllegalStateException -> ParsingError.ILLEGAL_VALUE
+          is InvalidFormatException -> ParsingError.ILLEGAL_VALUE
+          is IllegalArgumentException -> ParsingError.ILLEGAL_VALUE
+          is MismatchedInputException -> ParsingError.INVALID_TYPE
+          is InvalidTypeIdException -> ParsingError.INVALID_TYPE
+          is DateTimeParseException -> ParsingError.INVALID_FORMAT
```

Comment:
I realize the mapping here is not perfect, but it was tricky because exception types are not consistently used (e.g. we have validation that throws `IllegalStateException` for missing required fields), so I did the best I could. Please do suggest changes if you can think of better mappings.